File: app/src/main/java/com/googlenews/googlenews/view/NewsFragment.java
package com.googlenews.googlenews.view;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebResourceRequest;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.googlenews.googlenews.R;
import com.googlenews.googlenews.model.Article;

public class NewsFragment extends Fragment {
   static Article mArticle;
   static Context mContext;

    public NewsFragment() {
    }

   /* private NewsFragment(Context context, Article article) {
        this.mContext=context;
        this.mArticle=article;
        // Required empty public constructor
    }*/

    public static NewsFragment newInstance(Context context, Article article) {
        NewsFragment fragment = new NewsFragment();
        Bundle args = new Bundle();
        mContext=context;
        mArticle=article;
        fragment.setArguments(args);
        return fragment;
    }



    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view=inflater.inflate(R.layout.fragment_news, container, false);
        WebView mywebview = (WebView)view. findViewById(R.id.webView);
        WebSettings webSettings = mywebview.getSettings();
        webSettings.setJavaScriptEnabled(true);

        // Force links and redirects to open in the WebView instead of in a browser
        mywebview.setWebViewClient(new WebViewClient());
        mywebview.getSettings().setLoadsImagesAutomatically(true);
        mywebview.getSettings().setJavaScriptEnabled(true);
        mywebview.loadUrl(mArticle.getUrl());
        return view;
    }
}

File: app/src/main/java/com/googlenews/googlenews/view/NewsFragment.kt
package com.googlenews.googlenews.view

import android.content.Context
import android.os.Build
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient

import com.googlenews.googlenews.R
import com.googlenews.googlenews.model.Article

class NewsFragment : Fragment() {
     lateinit var mArticle: Article
     lateinit var mContext: Context

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.fragment_news, container, false)
        var mywebview : WebView? = null;
        mywebview = view.findViewById<View>(R.id.webView) as WebView
        var url=mArticle.url
        mywebview.settings.javaScriptEnabled = true
        mywebview.settings.loadsImagesAutomatically = true
        mywebview.setBackgroundColor(0)
       mywebview.webViewClient= WebViewClient()
        mywebview.loadUrl(url.toString())
        return view
    }

    companion object {
        fun  newInstance(context: Context, article: Article): NewsFragment {
            val fragment = NewsFragment()
            fragment.setVar(context,article)
            return fragment
        }
    }

    private fun setVar(context: Context, article: Article) {
        mArticle= article
        mContext=context
    }
}
